python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_graph/_digraph_group_chat.py

```diff
@@ -0,0 +1,615 @@
+
+                    else:
+                        # Case: unconditional edges — mark this source as completed for all its children
+                        target_nodes_names = [edge.target for edge in source_node.edges]
+
+                    for target in target_nodes_names:
+                        self._pending_execution[target].append(source)
+            else:
+                # TODO: Check if there are any usecase where the User can decide on the next speaker
+                pass
+
+        # After updating _pending_execution, check which nodes are now unblocked
+        for node_name in list(self._pending_execution):
+            if self._use_default_start and not self._default_start_executed:
+                if node_name == self._graph.default_start_node:
+                    next_speakers.add(node_name)
+                    self._default_start_executed = True
+                    break
+
+            if self._is_node_ready(node_name):
+                next_speakers.add(node_name)
+                node = self._graph.nodes[node_name]
+                if node.activation == "all":
+                    self._pending_execution.pop(node_name)
+                else:
+                    # If activation is any, remove the parent that just finished
+                    if source is not None:
+                        self._pending_execution[node_name] = [
+                            parent for parent in self._pending_execution[node_name] if parent != source
+                        ]
+
+                    # If none of the other parents of this node are active, remove this node from pending execution
+                    node_parents = self._parents[node_name]
+                    if not any(parent in self._active_nodes for parent in node_parents):
+                        self._pending_execution.pop(node_name)
+
+                if not many:
+                    break
+
+        # Prepopulate children of next_speakers into _pending_execution
+        for node_name in next_speakers:
+            for edge in self._graph.nodes[node_name].edges:
+                if edge.target not in self._pending_execution:
+                    self._pending_execution[edge.target] = []
+
+        # Mark newly selected speakers as active
+        for speaker in next_speakers:
+            if speaker not in self._active_nodes:
+                self._active_nodes.add(speaker)
+
+            self._active_node_count[speaker] += 1
+
+        if not self._pending_execution and not next_speakers and not self._active_nodes:
+            next_speakers = set([_DIGRAPH_STOP_AGENT_NAME])  # Call the termination agent
+
+        return list(next_speakers)
+
+    async def select_speakers(self, thread: List[BaseAgentEvent | BaseChatMessage]) -> List[str]:
+        return await self._select_speakers(thread)
+
+    async def select_speaker(self, thread: List[BaseAgentEvent | BaseChatMessage]) -> str:
+        """Select a speaker from the participants and return the
```

Comment:
We need to make a note here that multiple speaker selection will be supported in the future.